### PR TITLE
feat: update t0114

### PR DESCRIPTION
**Pull request by laurentsenta**
- [x] update t0114 to use gateway helper
- [x] add dnslink to t0114
- [x] add ipnslink to t0114

### CODE REVIEW

File: tooling/ipns/record.go [hunk 1]
```diff
@@ -41,10 +51,46 @@ func (i *IpnsRecord) Validity() time.Time {
...
+	// return s
+}
+
+func (i *IpnsRecord) IdV1() string {
+	return i.IntoCID(cid.Libp2pKey, mbase.Base36)
+}
 
-	return ipns.ValidateWithPeerID(id, i.pb)
+func (i *IpnsRecord) B58MH() string {
+	return i.id.String()
```
Comment (laurentsenta) [inline tooling/ipns/record.go:95]: Highlight: is this the correct implementation?

File: tooling/ipns/record.go [hunk 2]
```diff
@@ -41,10 +51,46 @@ func (i *IpnsRecord) Validity() time.Time {
...
+	panic("not a v0 id")
+	// This code won't work if key starts with "1", this is an identity multihash, which
+	// makes `NewCidV0` panic.
+	// c := peer.ToCid(i.id)
+	// s, err := cid.NewCidV0(c.Hash()).StringOfBase(mbase.Base58BTC)
+	// if err != nil {
+	// 	panic(err)
+	// }
+	// return s
+}
```
Comment (laurentsenta) [inline tooling/ipns/record.go:88]: Highlight: I couldn't find a reusable way to do this, so I copied some code out of kubo.

File: tooling/ipns/record.go (hunk 1 above)
Comment (lidel) [reply tooling/ipns/record.go:95]: cc @hacdias in context of https://github.com/ipfs/boxo/pull/339

File: tooling/ipns/record.go [hunk 3]
```diff
@@ -41,10 +51,39 @@ func (i *IpnsRecord) Validity() time.Time {
...
+	c = cid.NewCidV1(uint64(codec), c.Hash())
+	s, err := c.StringOfBase(base)
 	if err != nil {
-		return err
+		return "", err
 	}
+	return s, nil
+}
+
+func (i *IpnsRecord) IntoCID(codec multicodec.Code, base mbase.Encoding) string {
```
Comment (galargh) [inline tooling/ipns/record.go:67]: I'd find the following more natural given all the toStrings, etc. 

```suggestion
func (i *IpnsRecord) ToCID(codec multicodec.Code, base mbase.Encoding) string {
```

**Review by galargh — APPROVED** [review]

File: tooling/ipns/record.go (hunk 1 above)
Comment (laurentsenta) [reply tooling/ipns/record.go:88]: Thanks for the note, I logged that as an item in #8, we'll review once the PR is merged!

**Pull request merged**